### ISSUE TITLE
Animating `x` and `rotation` is buggy

### ISSUE THREAD

**Issue by basecode** [uxebu/bonsai#62]
Animating `x` and `rotation` moves the `Rect` instance too far. Seems it varies depending on the duration of the Animation. Using a Path instance instead works as expected (of course, the transform-origin is different).

```
var square = new Rect(0, 0, 100, 100);

// compared to a Path instance
//square = new Path('v100h100v-100');

square.addTo(stage);
square.fill('red');
square.animate('1s', {
  x: 300,
  rotation: Math.PI*2, 
  fillColor: 'green'
});

// line
new Path('M0,120h300v10h1v-20h-1v9h-300').fill('black').addTo(stage); 
```

**Comment by padolsey** [comment]
cc @davidaurelio

**Comment by davidaurelio** [comment]
Yes, this is a known problem. It is caused by multiple matrix transformations at once. bitmap-move-rotation.js shows the problem.

@basecode came up with other problems regarding rotation as well (offsets are changed), and I’m trying to figure out a way that makes it all work.

**Comment by padolsey** [comment]
@davidaurelio could I assign this to you? I don't think I have sufficient knowledge of this issue to attempt a fix.

**Comment by davidaurelio** [comment]
Sure

**Comment by iamdustan** [comment]
I’m attempting to understand the matrix transforms (and how graphics programming works) and I’ve been experimenting in Orbit. 

I came up with this <a href="http://orbit.bonsaijs.org/#%0Avar%20rect%20%3D%20new%20Rect(100%2C%20100%2C%20100%2C%20100).fill(%22red%22).addTo(stage)%3B%0Avar%20rect2%20%3D%20new%20Rect(100%2C%20100%2C%20100%2C%20100).fill(%22yellow%22).attr('opacity'%2C%200.8).addTo(stage)%3B%0A%0Avar%20star%20%3D%20new%20Star(100%2C100%2C25%2C5%2C3).fill('green').addTo(stage)%3B%0Avar%20star2%20%3D%20new%20Star(100%2C100%2C25%2C5%2C3).fill('orange').attr('opacity'%2C%200.8).addTo(stage)%3B%0A%0Avar%20center%20%3D%20new%20Rect(98%2C98%2C4%2C4).fill('black').addTo(stage)%0Avar%20startMatrix%20%3D%20new%20Matrix%3B%0Avar%20endMatrix%20%3D%20new%20Matrix%3B%0Avar%20rotation%20%3D%20toRadians(45)%3B%0Avar%20options%20%3D%20%7B%20repeat%3A%201000%20%7D%0A%0A%2F%2F%20it%20was%20at%2050%2C%2050%0Arect.attr('origin'%2C%20new%20Point(200%2C%20200))%3B%0A%2F%2F%20so%20I%20set%20it%20to%20200%2C%20200...not%20sure%20why%20it%20works%0A%0Avar%20transformOrigin%20%3D%20(new%20Matrix).transformPoint(rect.attr('origin'))%3B%0A%0Arect.animate(new%20KeyframeAnimation('3s'%2C%20%7B%0A%20%20'50%25'%3A%20%7B%20matrix%3A%20startMatrix.createBox(0.5%2C%200.5%2C%20rotation%2C%20transformOrigin.x%2C%20transformOrigin.y)%20%7D%2C%0A%20%20'100%25'%3A%20%7B%20matrix%3A%20endMatrix.createBox(1%2C%201%2C%200%2C%20100%2C%20100)%20%7D%0A%7D%2C%20options))%3B%0Arect2.animate(new%20KeyframeAnimation('3s'%2C%20%7B%0A%20%20'50%25'%20%3A%20%7B%20scale%3A%200.5%2C%20rotation%3A%20rotation%2C%20x%3A%20200%2C%20y%3A%20200%20%7D%2C%0A%20%20'100%25'%20%3A%20%7B%20scale%3A%201%2C%20rotation%3A%200%2C%20x%3A%20100%2C%20y%3A%20100%20%7D%0A%7D%2C%20options))%3B%0A%0AstartMatrix.createBox(0.5%2C%200.5%2C%20rotation%2C%20transformOrigin.x%2C%20transformOrigin.y)%0Astar.animate(new%20KeyframeAnimation('3s'%2C%20%7B%0A%20%20'50%25'%3A%20%7B%20matrix%3A%20startMatrix%20%7D%2C%0A%20%20'100%25'%3A%20%7B%20matrix%3A%20endMatrix%20%7D%0A%7D%2C%20options))%3B%0Astar2.animate(new%20KeyframeAnimation('3s'%2C%20%7B%0A%20%20'50%25'%3A%20%7B%20scale%3A%200.5%2C%20rotation%3A%20rotation%2C%20x%3A%20200%2C%20y%3A%20200%20%7D%2C%0A%20%20'100%25'%3A%20%7B%20scale%3A%201%2C%20rotation%3A%200%2C%20x%3A%20100%2C%20y%3A%20100%20%7D%0A%7D%2C%20options))%3B%0A%0Acenter.animate(new%20KeyframeAnimation('3s'%2C%20%7B%0A%20%20'50%25'%3A%20%7B%20x%3A%20200%2C%20y%3A%20200%20%7D%2C%0A%20%20'100%25'%3A%20%7B%20x%3A%20100%2C%20y%3A%20100%20%7D%0A%7D%2C%20options))%3B%0A%0Afunction%20toRadians(deg)%20%7B%0A%20%20return%20deg*Math.PI%2F180%0A%7D%0A%0A">Demo</a> that appears to make the Rect do the expected translation, but I’m not 100% positive why it all works.

Especially the origin update that I made. Thoughts? /cc @davidaurelio (since you’re assigned)

**Comment by iamdustan** [comment]
Oh, I also have a Star shape in the demo and doing an animation based on the matrix rather than the rotation also has it rotate at a different rate, oddly enough.

**Comment by iamdustan** [comment]
Bollocks. Attempting to do do animations with `matrix.createBox` fails. Try changing the degrees in that demo to 180. I would imagine that this wouldn’t happen with create box from this https://github.com/uxebu/bonsai/blob/master/test/display_object-spec.js#L152-159 comment...
